src/aiida_quantumespresso/workflows/xspectra/core.py

```diff
@@ -0,0 +1,910 @@
+        if unrecoverable_failures:
+            return self.exit_codes.ERROR_SUB_PROCESS_FAILED_XSPECTRA
+
+        self.ctx.all_lanczos_computed = True
+
+    def run_all_xspectra_plot(self):
+        """Run an `XspectraBaseWorkChain` for each 3-vector given for epsilon to plot the final spectra.
+
+        This part simply convolutes and plots the spectra from the already-computed Lanczos
+        of ``run_all_xspectra_plot``. Only run if requested via ``run_replot`` in the inputs.
+        """
+
+        finished_calculations = self.ctx.finished_lanczos
+
+        core_wfc_data = self.ctx.core_wfc_data
+
+        xspectra_plot_calcs = {}
+        for parent_xspectra in finished_calculations:
+            xspectra_inputs = AttributeDict(self.exposed_inputs(XspectraBaseWorkChain, 'xs_plot'))
+            # The epsilon vectors are not needed in the case of a replot, however they
+            # will be needed by the Parser at the end
+            xspectra_parameters = xspectra_inputs.xspectra.parameters.get_dict()
+
+            parent_output_dict = parent_xspectra.outputs.output_parameters.get_dict()
+            parent_calc_job = parent_xspectra.outputs.output_parameters.creator
+            eps_vector = parent_output_dict['xepsilon']
+            xspectra_parameters['INPUT_XSPECTRA']['xepsilon(1)'] = eps_vector[0]
+            xspectra_parameters['INPUT_XSPECTRA']['xepsilon(2)'] = eps_vector[1]
+            xspectra_parameters['INPUT_XSPECTRA']['xepsilon(3)'] = eps_vector[2]
+            old_label = parent_xspectra.label
+            new_label = old_label.replace('prod', 'plot')
+            xspectra_inputs.xspectra.parent_folder = parent_xspectra.outputs.remote_folder
+            xspectra_inputs.kpoints = parent_calc_job.inputs.kpoints
+            xspectra_inputs.xspectra.core_wfc_data = core_wfc_data
+            xspectra_inputs.metadata.label = new_label
+            xspectra_inputs.metadata.call_link_label = new_label
+
+            xspectra_inputs.xspectra.parameters = orm.Dict(dict=xspectra_parameters)
+
+            if self.ctx.dry_run:
+                return xspectra_inputs
+
+            future_xspectra = self.submit(XspectraBaseWorkChain, **xspectra_inputs)
+            self.report(
+                f'launching XspectraBaseWorkChain<{future_xspectra.pk}> for epsilon vector {eps_vector} (Replot)'
+            )
+            xspectra_plot_calcs[f'{new_label}'] = future_xspectra
+
+        return ToContext(**xspectra_plot_calcs)
+
+    def inspect_all_xspectra_plot(self):
+        """Verify that the `XspectraBaseWorkChain` re-plot sub-processes finished successfully."""
+
+        calculations = []
+        labels = self.ctx.xspectra_calc_labels
+        for label in labels:
+            calculation = self.ctx[f'{label}_plot']
+            calculations.append(calculation)
+        unrecoverable_failures = False
+
+        finished_replots = []
+        for calculation in calculations:
```

Comment:
Same here. Instead of manually having to retrieve a list of calculations through the labels, simply use the `append_` method to directly create a list of nodes in the context.